src/parquet/file/writer-internal.cc

```diff
@@ -148,16 +148,19 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
 }
 
 void RowGroupSerializer::Close() {
-  if (current_column_index_ != schema_->num_columns() - 1) {
-    throw ParquetException("Not all column were written in the current rowgroup.");
+  if (!closed_) {
```

Comment:
I'd prefer to have close idempotent. Thus you can call it explicitly in the code or implicitly in the destructor.